pkg/reconciler/revision/resources/deploy_test.go

```diff
@@ -219,22 +238,18 @@ var (
 			},
 		},
 	}
+)
 
-	defaultRevision = &v1.Revision{
```

Comment:
The existing `defaultRevision` supports single container and to add testcase for multiple container need to write another function just with the `podSpec` change.

So in order to avoid code duplication moved container section to another function and embedded `defaultRevision` in revision function

Reply:
An option could be to create a RevisionOption func to add the extra container to the existing revision.

ie. 

```go
func WithSidecar(c v1.Container) RevisionOption {
  return func (r v1.Revision) {
    r.Containers = append(r.Containers, c)
  // potential validation
  }
}
```

Reply:
Added `func withContainers(containers []corev1.Container) RevisionOption {` function as that contains containers for both serving and non-serving